scripts/compiling.py

```diff
@@ -0,0 +1,322 @@
+import lxml.etree as ET
+import sys
+import os
+
+# shorthand for yaml tabulation
+def getTabs(n):
+	str=""
+	for x in range(n):
+		str+="  "
+	return str
+
+def processSubProperty(prop):
+	if prop.fileName not in msgFiles:
+		subMsg = ""
+		for subProp in prop.properties:
+			if subProp.fileName is not None:
+				processSubProperty(subProp)
+			else:
+				if subProp.unit is not None and subProp.unit == "enum":
+
+					# sort enumeration values for readability
+					for value in sorted( ((v,k) for k,v in subProp.enumeration.iteritems())):
+						subMsg+=subProp.type+" "+value[1]+"="+str(value[0])+"\n"
+			subMsg+=formatProperty(subProp)
+		subFileName = prop.fileName+".msg"
+		msgFiles.append(prop.fileName)
+
+		text_file = open(subFileName, "w")
+		text_file.write(subMsg)
+		text_file.close()
+
+# separate the message generation for messages declared inside services
+def processMessage(module, topic):
```

Comment:
Same idea. Not returning any kind of feedback.

Reply:
https://github.com/erlerobot/HRIM/pull/16#discussion_r241386747